cloud/aws/deploy/website.go

```diff
@@ -397,51 +525,72 @@ func (a *NitricAwsPulumiProvider) deployCloudfrontDistribution(ctx *pulumi.Conte
 		return err
 	}
 
-	// apply invalidation on the distribution when files change
-	pulumi.All(a.Distribution.ID().ToStringOutput(), a.websiteChangedFileOutputs.ToStringArrayOutput()).ApplyT(func(args []interface{}) error {
-		cdnID := args[0].(string)
-		websiteChangedFileKeys := []string{}
-
-		// Filter out empty strings from the array
-		for _, key := range args[1].([]string) {
-			if key != "" {
-				websiteChangedFileKeys = append(websiteChangedFileKeys, key)
-			}
+	if domainName != "" {
+		_, err = route53.NewRecord(ctx, fmt.Sprintf("cdn-alias-record-%s", a.StackId), &route53.RecordArgs{
+			ZoneId: pulumi.String(hostedZoneId),
+			Type:   pulumi.String("A"),
+			Name:   pulumi.String(strings.Split(".", domainName)[0]),
+
+			Aliases: route53.RecordAliasArray{
+				route53.RecordAliasArgs{
+					Name:                 a.Distribution.DomainName,
+					ZoneId:               a.Distribution.HostedZoneId,
+					EvaluateTargetHealth: pulumi.Bool(false),
+				},
+			},
+		})
+		if err != nil {
+			return err
 		}
+	}
 
-		if len(websiteChangedFileKeys) > 0 {
-			cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithRegion(a.Region))
-			if err != nil {
-				return fmt.Errorf("failed to load AWS config: %w", err)
+	if !a.AwsConfig.Cdn.SkipCacheInvalidation {
+		// apply invalidation on the distribution when files change
+		pulumi.All(a.Distribution.ID().ToStringOutput(), a.websiteChangedFileOutputs.ToStringArrayOutput()).ApplyT(func(args []interface{}) error {
```

Comment:
I think all this validation stuff can just bust the entire cache, just use the md5 hashes to detect when to trigger it. See the Azure / GCP implementation.

This will remove the need to read the objects each deployment.